src/Types/TransactionStatus.cs

```diff
@@ -0,0 +1,80 @@
+using Concordium.Grpc.V2;
+using Concordium.Sdk.Exceptions;
+
+namespace Concordium.Sdk.Types;
+
+/// <summary>
+/// A common return type for TransactionStatus.
+/// </summary>
+public interface ITransactionStatus
```

Comment:
Are there any drawbacks from doing `TransactionStatus<T> where T : TransactionStatus<T>` and its variants `TransactionStatusCommitted : TransactionStatus<TransactionStatusCommitted>`?

Does using the interface not entail that one has to match on the type at runtime?

Reply:
Yes one has to match at runtime - I’m quite confident this is standard C#.